### PR TITLE
Feature/vrvu-71: Add Estimated Age UserInterface

### PR DESCRIPTION
**Pull request by sherrif10**
PR here   https://sd-vxnaid.atlassian.net/browse/VRVU-71   cc @jdolkowski  @druchniewicz 

This is basically a box for adding estimated age.
Screenshot 2024-07-17 140614

### CODE REVIEW

File: app/src/main/java/com/jnj/vaccinetracker/register/dialogs/BirthDatePickerDialog.kt [hunk 1]
```diff
@@ -129,8 +133,24 @@ class BirthDatePickerDialog(
...
+        switchIsBirthDateEstimated.setOnCheckedChangeListener { _, isChecked ->
             updateLayoutVisibility()
+            if (isChecked) {
+                editTextEstimatedAge.addTextChangedListener(object : TextWatcher {
+                    override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {}
+                    override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {}
+                    override fun afterTextChanged(s: Editable?) {
+                        val estimatedAge = s?.toString()?.toIntOrNull()
+                        if (estimatedAge != null) {
+                            numberPickerYears.value = estimatedAge
```
Comment (druchniewicz) [inline app/src/main/java/com/jnj/vaccinetracker/register/dialogs/BirthDatePickerDialog.kt:145]: how does it work? how does it affect the date?
for example estimatedAge = 5 and 5 is assigned to year picker or do I understand wrong?

File: app/src/main/java/com/jnj/vaccinetracker/register/dialogs/BirthDatePickerDialog.kt [hunk 2]
```diff
@@ -92,7 +96,7 @@ class BirthDatePickerDialog(
 
         if (selectedDate == null) {
             val year = c.get(Calendar.YEAR)
-            val month = c.get(Calendar.MONTH) + 1
+            val month = c.get(Calendar.MONTH)
```
Comment (druchniewicz) [inline app/src/main/java/com/jnj/vaccinetracker/register/dialogs/BirthDatePickerDialog.kt:99]: why it is changed?

**Review by druchniewicz — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/jnj/vaccinetracker/register/dialogs/BirthDatePickerDialog.kt (hunk 2 above)
Comment (sherrif10) [reply app/src/main/java/com/jnj/vaccinetracker/register/dialogs/BirthDatePickerDialog.kt:99]: i will revert this. Thanks

File: app/src/main/java/com/jnj/vaccinetracker/register/dialogs/BirthDatePickerDialog.kt (hunk 1 above)
Comment (sherrif10) [reply app/src/main/java/com/jnj/vaccinetracker/register/dialogs/BirthDatePickerDialog.kt:145]: ideally estimated Age should be assigned to the age value. as simple as that. Could you try to give me how you would implement this ?

Comment (druchniewicz) [reply app/src/main/java/com/jnj/vaccinetracker/register/dialogs/BirthDatePickerDialog.kt:145]: we don't have field in database for age. I would go with solution when estimated age is entered/changed then year picker is automatically set to correct year, e.g. we have now 2024, when we enter estimated age e.g. to 7 years, then year picker is automatically set to 2017 and this date is saved in database

Comment (sherrif10) [reply app/src/main/java/com/jnj/vaccinetracker/register/dialogs/BirthDatePickerDialog.kt:145]: This Makes sense. , Let me implement that kind of logic then .

**Comment by jdolkowski** [comment]
I think we should take a different approach here. The popup should remain unchanged. If date is estimated birth date should display calculated age instead of a date. cc @sherrif10

**Comment by sherrif10** [comment]
Okay i get, i will try to fix it.

**Comment by sherrif10** [comment]
Closing this, i will open a new PR

**Pull request closed without merging**